inst/include/Rcpp/algorithms.h

```diff
@@ -0,0 +1,74 @@
+namespace Rcpp {
+namespace algorithm {
+
+namespace helpers {
+    struct log {
+        inline double operator()(double val) {
```

Comment:
I wonder if we could make these more generic by:

1. Templating on `int RTYPE`,
2. Inferring the return + argument type with the `storage_type<>` type trait (https://github.com/RcppCore/Rcpp/blob/master/inst/include/Rcpp/traits/storage_type.h),
3. Use the appropriate `Vector<RTYPE>::is_na()` method.

Presumedly if someone tried to instantiate a bogus template it would just fail to compile but we should double check that as well. That said, this might 'leak' out into the top-level `log()` functions in an unnatural way.